src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -72,6 +72,18 @@ extends AbstractMap[A, B]
     else Some(e.value)
   }
 
+  override def getOrElseUpdate(key: A, defaultValue: => B): B = {
+    val i = index(elemHashCode(key))
+    val entry = findEntry0(key, i)
+    if (entry != null)
+      entry.value
+    else {
+      val newEntry = createNewEntry(key, defaultValue)
+      addEntry0(newEntry, i)
+      newEntry.value
+    }
+  }
```

Comment:
Oops, didn't read the signatures correctly.  It's not evaluating twice which is a problem; it's evaluating more than zero times when the entry exists.  You're right; we can't do it this way.